metals/src/main/scala/scala/meta/internal/builds/BuildTools.scala

```diff
@@ -227,12 +234,9 @@ final class BuildTools(
       Some(GradleBuildTool.name)
     else if (mavenProject.exists(MavenBuildTool.isMavenRelatedPath(_, path)))
       Some(MavenBuildTool.name)
-    else if (isMill && MillBuildTool.isMillRelatedPath(path))
+    else if (isMill && MillBuildTool.isMillRelatedPath(path) || isMillBsp(path))
       Some(MillBuildTool.name)
-    else if (
-      path.isFile && path.filename.endsWith(".json") &&
-      path.parent.filename == ".bsp"
-    )
+    else if (isInBsp(path))
```

Comment:
Maybe let's filter out `knowBsps` here. I feel like if `sbt.json` changes, we might not want to do anything, especially if we're connected to `bloop`.

Reply:
Wouldn't we want to reconnect at least in this case?

Reply:
I agree with @tgodzik, if we see `sbt.json` created I think we should react the same way as if this was a custom build tool (offer to connect)

Reply:
Sure, reconnect might make sense. If we don't use `bloop` but `sbt`. But it won't reconnect, it will just reload. We do reconnect only for custom bps now (look `Indexer#reloadWorkspaceAndIndex`).

Reply:
But if its a newly detected build tool won't it run `MetalsLspService.onBuildToolsAdded` and then in `slowConnectToBuildServer` call `quickConnect`?

Reply:
If it's newly detected it will. But that means there was no `sbt` previously in the workspace, so no `build.sbt`, and `.bsp/sbt.json` just got added. Now either `build.sbt` got added too, which is already detected by other cases, or just `.bsp/sbt.json` was added, which I don't think we can handle properly at the moment anyway. `.bsp/sbt.json` won't cause `sbt` to be detected as a build tool, and it won't be treated as a custom bps either (though we could do that but it's a rather odd case).

Reply:
You're right and I also don't see currently a way to handle changing `.bsp/sbt.json` . 

> Maybe let's filter out knowBsps here. I feel like if sbt.json changes, we might not want to do anything, especially if we're connected to bloop.

But if we are connected to `sbt` and `scala-cli.json` gets created, we won't get notified.

Since this PR doesn’t change anything it this behaviour, maybe we can work on it in a follow up?

Reply:
Sounds good to me!